src/docker/dockerfileParser.ts

```diff
@@ -0,0 +1,73 @@
+import * as fs from 'fs';
+import { CommandEntry, parse as rawDockerfileParser} from 'docker-file-parser';
+
+import { IDockerParser } from "./parser";
+import { shell, ShellResult } from "../shell";
+
+class RawDockerfile {
+    private commandEntries: CommandEntry[];
+
+    constructor(private readonly dockerfilePath: string) {
+        const dockerData = fs.readFileSync(dockerfilePath, 'utf-8');
+        this.commandEntries = rawDockerfileParser(dockerData, { includeComments: false });
+    }
+
+    public getCommandsOfType(...commands: string[]): CommandEntry[] {
+        return this.commandEntries.filter((entry) => {
+            const cmdName = entry.name.toLowerCase();
+            return commands.find((key) => key === cmdName);
+        });
+    }
+
+    public mergeCommandArgsOfType(command: string): string[] {
+        let args = [];
+        this.commandEntries.forEach((entry) => {
+            if (entry.name.toLowerCase() === command) {
+                args = args.concat(entry.args);
+            }
+        });
+        return args;
+    }
+
+    public searchInArgs(regularExpression: RegExp, scope?: string[]): RegExpMatchArray {

```

Comment:
Why is this called `scope` instead of `commands` as it was in `getCommandsOfType`?